xla/service/gpu/gpu_hlo_cost_analysis.cc

```diff
@@ -316,12 +336,65 @@ int64_t FlopsPerElement(const absl::string_view device_name,
   return FindOrDefault(dtype_profiles->get(), opcode, kDefaultFlopsPerElement);
 }
 
+int64_t GetFlopsForElementwiseOp(const GpuDeviceInfo* gpu_device_info,
+                                 const HloOpcode op_code, const Shape& shape) {
+  int64_t flop_per_element = FlopsPerElement(
+      gpu_device_info ? gpu_device_info->name : kDefaultDeviceName,
+      shape.element_type(), op_code);
+  return flop_per_element * ShapeUtil::ElementsInRecursive(shape);
+}
+
+int64_t GetFlopsForElementwiseOp(const GpuDeviceInfo* gpu_device_info,
+                                 const HloInstruction* instr) {
+  LOG(ERROR) << "#####ShapeUtil::ElementsInRecursive(instr->shape()) "
+             << ShapeUtil::ElementsInRecursive(instr->shape())
+             << " INSTR: " << instr->ToString();
+
+  return GetFlopsForElementwiseOp(gpu_device_info, instr->opcode(),
+                                  instr->shape());
+}
+
+Status GpuHloCostAnalysis::HandleAllReduce(const HloInstruction* allreduce) {
+  const HloModuleConfig& config = allreduce->GetModule()->config();
+  int64_t num_devices = config.num_partitions();
+  int64_t num_replicas = config.replica_count();
+  // Data parallel ranks
+  int64_t num_ranks = num_devices;
+
+  int64_t output_bytes_accessed = 0;
+  ShapeUtil::ForEachSubshape(
```

Comment:
May be a comment here to clarify that async-all reduce is handled here correctly as output shape is same as input shape. Because for other async collectives its different and the output shape includes both input and output shape

Reply:
added comment to clarify